### DISCUSSION TITLE
Root directory at 98%

### DISCUSSION THREAD

**Discussion by thegreekman** [Security-Onion-Solutions/securityonion#7578] [Unsupported Versions] [answered]
Hello, my root directory is at 98%. Its a 500G HardDrive. All the space is being taken up by data in /var/lib/docker/overlay2. Is there anything I Can delete to make more space?

**Comment by 0nnyx** [comment]
You might try `sudo so-docker-prune `

**Reply by thegreekman** [reply]
will that delete any vital data?

**Reply by 0nnyx** [reply]
No it's safe and will only remove old and unused docker images

**Reply by thegreekman** [reply]
Ok, this is the output of docker system df
![image](https://user-images.githubusercontent.com/27656903/158652698-1471d3de-ea6b-4137-b4d4-288b83ef8ed2.png)
IF i run docker system prune --volumes, that will remove all the unecessary volumes without any issues correct?

**Reply by 0nnyx** [reply]
You're safe to do docker system prune as long as all your docker containers are running since docker will not remove any element in use. Confirm first with so-status that all containers are in OK status before doing the prune.
I tried it on my installation and didn't get any problem.

**Reply by thegreekman** [reply]
Do you know why this directory is filling up so quickly? is that normal?

**Reply by dougburks** [reply]
Did you install using our Security Onion ISO image or did you install on top of an existing Ubuntu/CentOS installation?

If the latter, how did you partition the box?

Is this a standalone installation?  In general, 500GB isn't much for a standalone when you're trying to store Docker images, full packet capture, and Elasticsearch data.

**Reply by thegreekman** [reply]
Installed via Security Onion ISO image. This is a standalone installation. A 5TB ISCSI is attached for the /nsm directory(pcap and elasticsearch data). The only thing filling up the root directory is the docker images but this hasn't happened in the past, at least not this fast. 300G is partitioned for / directory.

**Reply by dougburks** [reply]
Typically, the box should just keep the current version of the Docker images and the previous version. Checking one of my boxes, this only takes up 40GB of space:
```
du --max-depth=1 -h /var/lib/docker/
236M	/var/lib/docker/containers
0	/var/lib/docker/plugins
39G	/var/lib/docker/overlay2
112M	/var/lib/docker/image
85M	/var/lib/docker/volumes
0	/var/lib/docker/trust
172K	/var/lib/docker/network
0	/var/lib/docker/swarm
16K	/var/lib/docker/builder
92K	/var/lib/docker/buildkit
0	/var/lib/docker/tmp
0	/var/lib/docker/runtimes
40G	/var/lib/docker/
```

For comparison to your `docker system df` screenshot, here's mine:
```
docker system df
TYPE            TOTAL     ACTIVE    SIZE      RECLAIMABLE
Images          66        23        28.49GB   19.61GB (68%)
Containers      24        24        216.8MB   0B (0%)
Local Volumes   223177    4         203.5kB   203.4kB (99%)
Build Cache     0         0         0B        0B
```

Your screenshot shows 218GB in `Local Volumes` whereas mine is only 203KB. Are you using any custom Docker images or configuration?

You might also want to consider the `sudo docker system prune -a` command as mentioned at:
https://docs.securityonion.net/en/2.3/soup.html#security-onion-version-updates

**Reply by thegreekman** [reply]
Thanks, i did sudo docker system prune --volumes and it fixed the issue. I then had to restart all the containers because logstash
stopped ingesting logs, Im assuming because several of the services broke when the disk space became full.

**Comment by thegreekman** [comment, marked as answer]
Resolved

**Reply by cyb3rz3us** [reply]
[Helper link to the actual answer.](https://github.com/Security-Onion-Solutions/securityonion/discussions/7578#discussioncomment-2488110)
